### ISSUE TITLE
[Question]: Authentication through Custom Chrome Tab on Android

### ISSUE THREAD

**Issue by MatthiasKroiss** [supabase-community/supabase-kt#297]
### General info

- [X] I checked the [troubleshooting](https://github.com/supabase-community/supabase-kt/wiki/Troubleshooting) page for similar problems

## What I already have
Hi, I have a webapp (SvelteKit) that uses Supabase auth. Now I want to create a native Android app in kotlin. 

## Question
I had the idea to use my already existing website for auth, but how can I do so using the supabase-kt library. Is there already a way I have not found in the docs to open a custom chrome tab and proceed with auth (on my site, with email, and github) and get back to my android app as authenticated user to use the kotlin library for database calls, etc.

If there is no solution I would still be very happy about advice.

Thank you.

**Comment by jan-tennert** [comment]
There is a config option to use custom tabs for GoTrue, do you want to do all of the auth stuff in a custom tab via your website?

**Comment by MatthiasKroiss** [comment]
Yes I want to do all auth stuff in one place on my website.

**Comment by jan-tennert** [comment]
So the built-in functionality for using Custom Tabs is for opening OAuth flows and then navigate back to the App via deeplinks. Doing all the authentication in a Custom Tab is certainly not impossible, but I'd advise to rather use in-app auth as you'd have to do everything yourself, and I'm not sure if it's the best UX for users. Is there a reason you don't want to create your own form in-app?

**Comment by MatthiasKroiss** [comment]
It was an idea, so that I don't have to maintain 2 codebases if e.g a new OAuth provider gets added. But then i will stick to the native in-app form.

**Comment by jan-tennert** [comment]
> It was an idea, so that I don't have to maintain 2 codebases if e.g a new OAuth provider gets added. But then i will stick to the native in-app form.

Makes sense, can't think of any good solution for that. If you have any more questions let me know!
